artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/jdbc/ActiveMQScheduledLeaseLock.java

```diff
@@ -84,37 +89,55 @@ public synchronized void stop() {
    }
 
    @Override
-   public void run() {
+   public synchronized void run() {
+      if (!isStarted()) {
+         return;
+      }
       final long lastRenewStart = this.lastLockRenewStart;
       final long renewStart = System.nanoTime();
+      boolean lockLost = true;
       try {
-         if (!this.lock.renew()) {
-            ioCriticalErrorListener.onIOException(new IllegalStateException(lockName + " lock can't be renewed"), "Critical error while on " + lockName + " renew", null);
-         }
+         lockLost = !this.lock.renew();
       } catch (Throwable t) {
-         ioCriticalErrorListener.onIOException(t, "Critical error while on " + lockName + " renew", null);
-         throw t;
+         LOGGER.warnf(t, "%s lock renew has failed", lockName);
+         if (lock.localExpirationTime() > 0) {
+            final long millisBeforeExpiration = (lock.localExpirationTime() - System.currentTimeMillis());
+            // there is enough time to retry to renew it?
+            if (millisBeforeExpiration >= this.renewPeriodMillis) {
+               lockLost = false;
+            }
+         }
+      }
+      // a failed attempt to renew is treated as a lost lock
+      if (lockLost) {
+         try {
+            lockListener.lostLock();
+         } catch (Throwable t) {
+            LOGGER.warnf(t, "Errored while notifying %s lock listener", lockName);
```

Comment:
Can an error on handle the lost lock lead to a split brain? Is it a critical error in this case?

Reply:
That's a good question: I've added this one to track OOM with a meaningful msg, but in theory a lost lock should just close forcibly the Node manager to allow any pending use of it to stop as well

Reply:
We should always stop on this, but if we get OOM there is a risk to split brain because we cannot create a new thread to stop the broker as expected